### DISCUSSION TITLE
mdx support

### DISCUSSION THREAD

**Discussion by 7flash** [vercel/satori#551] [Ideas]
# Feature Request

## Is your feature request related to a problem? Please describe.
While Satori has impressive capabilities in translating HTML and CSS to SVG, there's an emerging need for richer content presentations like MDX, which combines Markdown and JSX.

## Describe the solution you'd like
It would be powerful if Satori could directly support rendering MDX content.

## Describe alternatives you've considered
While we can always pre-process MDX to plain HTML and then feed it to Satori, having native support would streamline the process, make it more efficient, and open up new possibilities for content presentation.

## Additional context
https://github.com/mdx-js/mdx

**Comment by shuding** [comment]
I think for now, this is out of the scope of this project. The Satori core is mostly a CSS layout → SVG translator and we will be focusing on that. This is because that part is already a lot of work (too many CSS details and we have typography issues like RTL).

Like you said, you can use another project transforms MDX → HTML structured data with CSS styles and then use Satori.

I'll convert this to a discussion thread and open to hear more feedback! Thanks.
